docs/articles/oso/how-to-open-source-software.rst

```diff
@@ -0,0 +1,28 @@
+:orphan:
+
+How to open source software
+===========================
+
+.. tagslist:: howto
+
+When open sourcing source code there are some steps that needs to be completed.
+The purpose of these steps are to ensure that the code has a sound license, the
```

Comment:
Don't we want to say "Publishing source code has some best practices associated with it along with license obligations."  These two points would cover, I believe, our practice of adding a "CONTRIBUTING.md" as well as demonstrate our understanding that many of the licenses we use *require* "complete and corresponding" source code. At the end of the day a CONTRIBUTING.md file is optional while a LICENSE is not. Shouldn't we firmly ground our HowTo in our understanding of compliance and praxis?

Reply:
You are correct in that some is optional some is not etc. However, from our side I would say both are still mandatory. So I do not think it matters to much to state why some are more relevant than others.

Since we are talking about a source code repository the point of complete and corresponding source code breaks down a bit. This is not a publication of a binary, we are just publishing the actual source code. You are indeed correct in that such text is required somewhere although I do not feel that this is the right context. For instance I have another PR about publishing binaries.

Reply:
Yeah keep in mind that this is intended to be a checklist to use when opening up code. There needs to be many more texts describing some principles and processes, but we want a How-To to have high cohesion and low coupling for maintenance purposes. Essentially a How-To should be referenced from "above" abstraction wise and almost never the other way around.  think it's OK to reference the OSO since that's difficult to get away from and will always be part of the Blueprint.

Reply:
I think we should consider if this is really a PELUX Baseline SWF topic. The exact checklist is likely to be different in using projects.